astropy/stats/funcs.py

```diff
@@ -761,6 +766,8 @@ def median_absolute_deviation(a, axis=None):
     # returns an masked array even if the result should be scalar). (#4658)
     if isinstance(a, np.ma.MaskedArray):
         func = np.ma.median
+    elif ignore_nan:
+        func = np.nanmedian
```

Comment:
I think this would fail if an `axis` argument is given because boolean indexing returns a 1d array